pkg/tests/end_to_end_tests/metric_ingest_bench_test.go

```diff
@@ -74,3 +76,29 @@ func BenchmarkMetricIngest(b *testing.B) {
 		b.StopTimer()
 	})
 }
+
+func BenchmarkNewSeriesIngestion(b *testing.B) {
+	seriesGen, err := testsupport.NewSeriesGenerator(100, 100, 4)
+	require.NoError(b, err)
+
+	requests := seriesGen.GetWriteRequests(2000)
+
+	withDB(b, "bench_e2e_new_series_ingest", func(db *pgxpool.Pool, t testing.TB) {
+		metricsIngestor, err := ingestor.NewPgxIngestorForTests(pgxconn.NewPgxConn(db), &ingestor.Cfg{
+			NumCopiers:              8,
+			InvertedLabelsCacheSize: cache.DefaultConfig.InvertedLabelsCacheSize,
+		})
+		require.NoError(b, err)
+		defer metricsIngestor.Close()
+
+		b.ResetTimer()
+		b.ReportAllocs()
+		for i := 0; i < len(requests); i++ {
```

Comment:
@Harkishen-Singh BTW what are the advantage when we ingest in batches? Don't we already batch in our implementation?

Reply:
I was trying to make it like what Prometheus does, sending new series in batches. But we could as well ingest all series in a single batch of 10000 like in here.